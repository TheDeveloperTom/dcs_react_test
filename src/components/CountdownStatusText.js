import React from "react";
import { Text } from "./CountdownStyles";

export default ({ halfway, remainingTime }) => {
  if (remainingTime === 0) {
    return <Text red>Timeâ€™s up!</Text>;
  }

  if (remainingTime <= halfway) {
    return (
      <Text blinking={remainingTime < 10} red={remainingTime < 20}>
        More than halfway there!
      </Text>
    );
  }
  return null;
};
