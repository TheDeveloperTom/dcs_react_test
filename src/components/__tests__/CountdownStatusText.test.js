import React from "react";
import CountdownStatusText from "../CountdownStatusText";

describe("CountdownStatusText", () => {
  const props = {
    halfway: 10,
    remainingTime: 20
  };
  const component = mount(<CountdownStatusText {...props} />);

  it("renders properly", () => {
    expect(component).toMatchSnapshot();
  });

  it("renders correct text when remainingTime === 0", () => {
    const nextProps = {
      ...props,
      remainingTime: 0
    };

    const component = mount(<CountdownStatusText {...nextProps} />);

    expect(
      component
        .find("Text")
        .at(0)
        .prop("red")
    ).toEqual(true);
  });

  it("it starts blinking when remaining time > halfway", () => {
    const nextProps = {
      ...props,
      remainingTime: 5
    };

    const component = mount(<CountdownStatusText {...nextProps} />);

    expect(
      component
        .find("Text")
        .at(0)
        .prop("blinking")
    ).toEqual(true);
  });

  it("doesn't renders when remaining time > halfway", () => {
    const component = mount(<CountdownStatusText {...props} />);

    expect(component.find("Text")).toHaveLength(0);
  });
});
