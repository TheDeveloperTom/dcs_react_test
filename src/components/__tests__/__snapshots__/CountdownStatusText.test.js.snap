// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountdownStatusText renders properly 1`] = `ReactWrapper {}`;
